plugins/reporters/evaluated-model/src/main/kotlin/EvaluatedModelMapper.kt

```diff
@@ -492,15 +492,18 @@ internal class EvaluatedModelMapper(private val input: ReporterInput) {
             visitedNodes += getInternalId() to createDependencyNode(dependency, linkage, issues)
 
             val children = visitDependencies { dependencies ->
-                dependencies.sortedBy { it.id }.map { node ->
-                    val nodeId = node.getInternalId()
-
-                    if (deduplicateDependencyTree && nodeId in visitedNodes) {
-                        // Cut the duplicate subtree here and only return the node without its children.
-                        visitedNodes.getValue(nodeId).copy(children = emptyList())
-                    } else {
-                        node.toEvaluatedTreeNode(scope, path + dependency)
-                    }
+                dependencies
```

Comment:
Just wondering, is the problem maybe (also) related to the `dependencies` sequence being sorted and mapped lazily? In other words, would moving the final `.toList()` to the beginning also fix the problem:

                dependencies.toList().sortedBy { it.id }.map { node ->
                    val nodeId = node.getInternalId()

                    if (deduplicateDependencyTree && nodeId in visitedNodes) {
                        // Cut the duplicate subtree here and only return the node without its children.
                        visitedNodes.getValue(nodeId).copy(children = emptyList())
                    } else {
                        node.toEvaluatedTreeNode(scope, path + dependency)
                    }
                }

Reply:
Turning this into a list first still exhibits the exact same problem, that only a single instance is used for all list elements. That instance changes it's data which in this case happens already during the `toList()` call.